core/Symbols.cc

```diff
@@ -960,13 +960,16 @@ void Symbol::recordSealedSubclass(MutableContext ctx, SymbolRef subclass) {
     ENFORCE(subclass.exists(), "Can't record sealed subclass for {} when subclass doesn't exist", this->show(ctx));
     ENFORCE(subclass.data(ctx)->isClassOrModule(), "Sealed subclass {} must be class", subclass.show(ctx));
 
+    auto selfRef = this->ref(ctx);
```

Comment:
Can you comment this? We have in the past looked at completely removing `ref`, and this is one that someone could easily come along not knowing the reasoning and inline this into the one usage below.

Reply:
Should we comment all the uses of `ref` in this file? It seems like most methods end up using `this->ref(gs)` to avoid the same bug, perhaps this is why @DarkDimius was suggesting we always pass in the `SymbolRef` to these methods?

Reply:
Either way works for me.